src/main/java/ArrayMinHeap.java

```diff
@@ -26,7 +26,41 @@ public ArrayMinHeap(int initialCapacity) {
 
     @Override
     public boolean add(T element) {
-        return false;
+        if (size() == heap.length) {
+            expandCapacity();
+        }
+        heap[size] = element;
+        bubbleUp(size);
+        size++;
+        return true;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void expandCapacity() {
+        T[] newHeap = (T[]) new Comparable[heap.length * 2];
+        System.arraycopy(heap, 0, newHeap, 0, heap.length);
```

Comment:
Don't use `arraycopy`. [Arrays#copyOf](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Arrays.html#copyOf(T%5B%5D,int)) exists for exactly this purpose.

Reply:
I ended up
1. removing `expandCapacity` altogether for just the `Arrays#copyOf`
2. Added a test for `WhenLarge`